volsupervisor/volsupervisor.go

```diff
@@ -42,8 +46,33 @@ retry:
 	go dc.watchAndSetGlobal(globalChan)
 	go info.HandleDebugSignal()
 
+	stopChan, err := lock.NewDriver(dc.Config).AcquireWithTTLRefresh(&config.UseVolsupervisor{Hostname: dc.Hostname}, dc.Global.TTL, dc.Global.Timeout)
+	if err != nil {
+		logrus.Fatal("Could not start volsupervisor: already in use")
+	}
+
+	sigChan := make(chan os.Signal, 1)
+
+	go func() {
+		<-sigChan
+		logrus.Infof("Removing volsupervisor global lock; waiting %v for lock to clear", dc.Global.TTL)
+		stopChan <- struct{}{}
+		time.Sleep(dc.Global.TTL + 1*time.Second) // give us enough time to try to clear the lock
+		os.Exit(0)
+	}()
+
+	signal.Notify(sigChan, syscall.SIGTERM, syscall.SIGINT)
+
 	dc.signalSnapshot()
-	dc.updatePolicies()
+	dc.updateVolumes()
+	// doing it here ensures the goroutine is created when the first poll completes.
+	go func() {
+		for {
+			time.Sleep(1 * time.Second)
```

Comment:
I don't care either way in this case

For loops which always sleep a fixed duration of time, I prefer `Tick()` because it communicates that's it's going to be sleeping right in the loop declaration rather than you seeing an infinite loop and having to look for a `Sleep()` somewhere in the body.  That's just personal preference, and I don't think it matters at all for tiny loops.